src/EditorFeatures/Core/Implementation/EditorLayerExtensionManager.cs

```diff
@@ -30,28 +32,61 @@ internal class EditorLayerExtensionManager : IWorkspaceServiceFactory
         public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
         {
             var optionService = workspaceServices.GetService<IOptionService>();
-            return new ExtensionManager(optionService, _errorHandlers);
+            var errorReportingService = workspaceServices.GetService<IErrorReportingService>();
+            var documentTrackingService = workspaceServices.GetService<IDocumentTrackingService>();
+            var errorLoggerService = workspaceServices.GetService<IErrorLogger>();
+            return new ExtensionManager(optionService, errorReportingService, documentTrackingService, errorLoggerService,  _errorHandlers);
         }
 
-        private class ExtensionManager : AbstractExtensionManager
+        internal class ExtensionManager : AbstractExtensionManager
         {
             private readonly List<IExtensionErrorHandler> _errorHandlers;
             private readonly IOptionService _optionsService;
+            private readonly IDocumentTrackingService _documentTrackingService;
+            private readonly IErrorReportingService _errorReportingService;
+            private readonly IErrorLogger _errorLoggerService;
 
-            public ExtensionManager(IOptionService optionsService, List<IExtensionErrorHandler> errorHandlers)
+            public ExtensionManager(
+                IOptionService optionsService,
+                IErrorReportingService errorReportingService,
+                IDocumentTrackingService documentTrackingService,
+                IErrorLogger errorLoggerService,
+                List<IExtensionErrorHandler> errorHandlers)
             {
                 _optionsService = optionsService;
                 _errorHandlers = errorHandlers;
+                _errorReportingService = errorReportingService;
+                _documentTrackingService = documentTrackingService;
+                _errorLoggerService = errorLoggerService;
             }
 
             public override void HandleException(object provider, Exception exception)
             {
-                if (_optionsService.GetOption(ExtensionManagerOptions.DisableCrashingExtensions))
+                if (provider is CodeFixProvider || provider is FixAllProvider)
```

Comment:
What about CodeRefactoringProvider? We need to have same behavior for fixes and refactorings.